tests/component/test_trigger_properties.py

```diff
@@ -0,0 +1,118 @@
+        yield task
+
+
+def test__trigger__get_float_property__returns_default_value(ai_voltage_task: Task):
+    """Test to validate getter for trigger property of float type."""
+    assert ai_voltage_task.triggers.start_trigger.dig_edge_dig_fltr_timebase_rate == 0.0
+
+
+def test__trigger__set_float_property__returns_assigned_value(ai_voltage_task: Task):
+    """Test to validate setter for trigger property of float type."""
+    value_to_test = 2.505
+    ai_voltage_task.triggers.start_trigger.dig_edge_dig_fltr_timebase_rate = value_to_test
+
+    assert ai_voltage_task.triggers.start_trigger.dig_edge_dig_fltr_timebase_rate == value_to_test
+
+
+def test__trigger__reset_float_property__returns_default_value(ai_voltage_task: Task):
+    """Test to validate resetting trigger property of float type."""
+    ai_voltage_task.triggers.start_trigger.dig_edge_dig_fltr_timebase_rate = 1.2
+
+    del ai_voltage_task.triggers.start_trigger.dig_edge_dig_fltr_timebase_rate
+
+    assert ai_voltage_task.triggers.start_trigger.dig_edge_dig_fltr_timebase_rate == 0.0
+
+
+def test__trigger__get_string_property__returns_default_value(ai_voltage_task: Task):
+    """Test to validate getter for trigger property of string type."""
+    assert ai_voltage_task.triggers.start_trigger.dig_edge_dig_fltr_timebase_src == ""
+
+
+def test__trigger__set_string_property__returns_assigned_value(ai_voltage_task: Task):
+    """Test to validate setter for trigger property of string type."""
+    value_to_test = "Test Value for Digital Edge Digital Filter Timebase Source"
+    ai_voltage_task.triggers.start_trigger.dig_edge_dig_fltr_timebase_src = value_to_test
+
+    assert ai_voltage_task.triggers.start_trigger.dig_edge_dig_fltr_timebase_src == value_to_test
+
+
+def test__trigger__reset_string_property__returns_default_value(ai_voltage_task: Task):
+    """Test to validate resetting trigger property of string type."""
+    ai_voltage_task.triggers.start_trigger.dig_edge_dig_fltr_timebase_src = "PFI3"
+
+    del ai_voltage_task.triggers.start_trigger.dig_edge_dig_fltr_timebase_src
+
+    assert ai_voltage_task.triggers.start_trigger.dig_edge_dig_fltr_timebase_src == ""
+
+
+def test__trigger__get_enum_property__returns_default_value(ai_voltage_task: Task):
+    """Test to validate getter for trigger property of enum type."""
+    assert ai_voltage_task.triggers.start_trigger.trig_type == TriggerType.NONE
+
+
+def test__trigger__set_trig_type_without_cfg_samp_clk__throws_daqerror(ai_voltage_task: Task):
+    """Test to validate error while setting trigger type without configuring sample clock."""
+    value_to_test = TriggerType.ANALOG_EDGE
+    ai_voltage_task.triggers.start_trigger.trig_type = value_to_test
+
+    with pytest.raises(DaqError) as exc_info:
+        _ = ai_voltage_task.triggers.start_trigger.trig_type
+
+    assert exc_info.value.error_type == DAQmxErrors.TRIG_WHEN_ON_DEMAND_SAMP_TIMING
```

Comment:
The test name has "set_trig_type" in the "when" clause, but the test actually sets it in the "arrange" section and gets it in the "act" section.

Suggestion:
- Move the set call to the "act" section. 
- The error is really caused by the set call, but it isn't reported until you verify the task. Instead of using a get call to verify the task, use `task.control(TaskMode.VERIFY)`
- Move "without_cfg_samp_clk" to the "given" clause: `test__ai_task_without_cfg_samp_clk__set_trig_type__throws_daqerror`

Reply:
Used  `ai_voltage_task.control(TaskMode.TASK_VERIFY)` instead of set call.

Updated the test name to `test__ai_task_without_cfg_samp_clk__set_trig_type__throws_daqerror`

Reply:
"Move the set call to the "act" section" means put it in the middle group of lines.

```
def test__trigger__set_trig_type_without_cfg_samp_clk__throws_daqerror(ai_voltage_task: Task):
    """Test to validate error while setting trigger type without configuring sample clock."""
    value_to_test = TriggerType.ANALOG_EDGE

    ai_voltage_task.triggers.start_trigger.trig_type = value_to_test
    with pytest.raises(DaqError) as exc_info:
        _ = ai_voltage_task.triggers.start_trigger.trig_type

    assert exc_info.value.error_type == DAQmxErrors.TRIG_WHEN_ON_DEMAND_SAMP_TIMING
```

Reply:
Updated as suggested